Source/Diagnostics/ReducedDiags/BeamRelevant.cpp

```diff
@@ -0,0 +1,376 @@
+            ofs << "[19]emittance_z(m)";  ofs << std::endl;
+            #elif (AMREX_SPACEDIM == 2)
+            ofs << "#";
+            ofs << "[1]step";             ofs << m_sep;
+            ofs << "[2]time(s)";          ofs << m_sep;
+            ofs << "[3]x_mean(m)";        ofs << m_sep;
+            ofs << "[4]z_mean(m)";        ofs << m_sep;
+            ofs << "[5]px_mean(kg m/s)";  ofs << m_sep;
+            ofs << "[6]py_mean(kg m/s)";  ofs << m_sep;
+            ofs << "[7]pz_mean(kg m/s)";  ofs << m_sep;
+            ofs << "[8]gamma_mean";       ofs << m_sep;
+            ofs << "[9]x_rms(m)";         ofs << m_sep;
+            ofs << "[10]z_rms(m)";        ofs << m_sep;
+            ofs << "[11]px_rms(kg m/s)";  ofs << m_sep;
+            ofs << "[12]py_rms(kg m/s)";  ofs << m_sep;
+            ofs << "[13]pz_rms(kg m/s)";  ofs << m_sep;
+            ofs << "[14]gamma_rms";       ofs << m_sep;
+            ofs << "[15]emittance_x(m)";  ofs << m_sep;
+            ofs << "[16]emittance_z(m)";  ofs << std::endl;
+            #endif
+            // close file
+            ofs.close();
+        }
+    }
+
+}
+// end constructor
+
+// function that compute beam relevant quantities
+void BeamRelevant::ComputeDiags (int step)
+{
+
+    // Judge if the diags should be done
+    if ( (step+1) % m_freq != 0 ) { return; }
+
+    // get MultiParticleContainer class object
+    auto & mypc = WarpX::GetInstance().GetPartContainer();
+
+    // get number of species (int)
+    auto nSpecies = mypc.nSpecies();
+
+    // get species names (std::vector<std::string>)
+    auto species_names = mypc.GetSpeciesNames();
+
+    // inverse of speed of light squared
+    auto inv_c2 = 1.0 / (PhysConst::c * PhysConst::c);
+
+    // If 2D-XZ, p.pos(1) is z, rather than p.pos(2).
+    #if (AMREX_SPACEDIM == 3)
+    int index_z = 2;
+    #elif (AMREX_SPACEDIM == 2)
+    int index_z = 1;
+    #endif
+
+    // loop over species
+    for (int i_s = 0; i_s < nSpecies; ++i_s)
+    {
+
+        // only beam species does
+        if (species_names[i_s] != m_beam_name) { continue; }
+
+        // get WarpXParticleContainer class object
+        auto & myspc = mypc.GetParticleContainer(i_s);
+
+        // get mass (Real)
+        auto m = myspc.getMass();
+
+        using PType = typename WarpXParticleContainer::SuperParticleType;
+
+        // weight sum
+        auto w_sum = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::w); });
+
+        // reduced sum over mpi ranks
+        ParallelDescriptor::ReduceRealSum(w_sum);
+
+        if (w_sum < std::numeric_limits<Real>::min() )
+        {
+            for (int i = 0; i < m_data.size(); ++i) { m_data[i] = 0.0; }
+            return;
+        }
+
+        // x mean
+        auto x_mean = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.pos(0) * p.rdata(PIdx::w) / w_sum; });
+
+        #if (AMREX_SPACEDIM == 3)
```

Comment:
```suggestion
#if (AMREX_SPACEDIM == 3)
```
We put all our pre-processor directives without indentation in the rest of the code. Could you also do it in this PR?